solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -2747,6 +2749,85 @@ impl tanssi_initializer::Config for Runtime {
     type SessionHandler = OwnApplySession;
 }
 
+pub struct BabeCurrentBlockRandomnessGetter;
+impl BabeCurrentBlockRandomnessGetter {
+    fn get_block_randomness() -> Option<[u8; 32]> {
+        Babe::author_vrf_randomness()
+    }
+
+    fn get_block_randomness_mixed(subject: &[u8]) -> Option<Hash> {
+        Self::get_block_randomness()
+            .map(|random_hash| mix_randomness::<Runtime>(random_hash, subject))
+    }
+}
+
+/// Combines the vrf output of the previous block with the provided subject.
+/// This ensures that the randomness will be different on different pallets, as long as the subject is different.
+fn mix_randomness<T: frame_system::Config>(vrf_output: [u8; 32], subject: &[u8]) -> T::Hash {
+    let mut digest = Vec::new();
+    digest.extend_from_slice(vrf_output.as_ref());
+    digest.extend_from_slice(subject);
+
+    T::Hashing::hash(digest.as_slice())
+}
+
+/// Read full_rotation_period from pallet_configuration
+pub struct ConfigurationCollatorRotationSessionPeriod;
+
+impl Get<u32> for ConfigurationCollatorRotationSessionPeriod {
+    fn get() -> u32 {
+        CollatorConfiguration::config().full_rotation_period
+    }
+}
+
+pub struct BabeGetRandomnessForNextBlock;
+
+impl GetRandomnessForNextBlock<u32> for BabeGetRandomnessForNextBlock {
+    fn should_end_session(n: u32) -> bool {
+        n != 1 && {
+            let diff = Babe::current_slot()
+                .saturating_add(1u64)
+                .saturating_sub(Babe::current_epoch_start());
+            *diff >= Babe::current_epoch().duration
+        }
+    }
+
+    fn get_randomness() -> [u8; 32] {
+        let block_number = System::block_number();
+        let random_seed = if block_number != 0 {
+            if let Some(random_hash) = {
+                BabeCurrentBlockRandomnessGetter::get_block_randomness_mixed(b"CollatorAssignment")
+            } {
+                // Return random_hash as a [u8; 32] instead of a Hash
+                let mut buf = [0u8; 32];
+                let len = sp_std::cmp::min(32, random_hash.as_ref().len());
+                buf[..len].copy_from_slice(&random_hash.as_ref()[..len]);
+
+                buf
+            } else {
+                // If there is no randomness (e.g when running in dev mode), return [0; 32]
+                // TODO: smoke test to ensure this never happens in a live network
+                [0; 32]
+            }
+        } else {
+            // In block 0 (genesis) there is randomness
```

Comment:
This typo is probably mine

```suggestion
            // In block 0 (genesis) there is no randomness
```

Reply:
fixed